### PR TITLE
chore(limit): add and remove transaction limits for individual wallets

### PR DESCRIPTION
**Pull request by Geoslim**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Does this PR introduce a breaking change?
- [ ] Yes
- [ ] No

### CODE REVIEW

File: routes/cpcash.php [hunk 1]
```diff
@@ -18,6 +18,9 @@
 Route::post('lock/{walletId}', [CPCashController::class, 'lockWallet']);
 Route::post('unlock/{walletId}', [CPCashController::class, 'unlockWallet']);
 
+Route::post('add-limit/{walletId}', [CPCashController::class, 'addWalletLimit']);
```
Comment (dammy001) [inline routes/cpcash.php:21]: These remove-limit routes are not required in this package.

File: src/Http/Requests/LimitRequest.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?php
```
Comment (dammy001) [inline src/Http/Requests/LimitRequest.php:1]: This is not required here.

File: routes/cpcash.php (hunk 1 above)
Comment (dammy001) [reply routes/cpcash.php:21]: Why are we adding these routes here?

File: src/Http/Controllers/CPCashController.php [hunk 3]
```diff
@@ -164,6 +165,25 @@ public function unlockWallet($walletId): JsonResponse
     {
         return $this->successResponse(CPCash::unlockWallet($walletId), trans('cpcash::wallet.unlock'));
     }
+	
+    /**
+     * @param int|string $walletId
+     * @param LimitRequest $request
+     * @return JsonResponse
+     */
+    public function addWalletLimit($walletId, LimitRequest $request): JsonResponse
```
Comment (dammy001) [inline src/Http/Controllers/CPCashController.php:174]: Why are we adding these methods to this package?

File: src/Services/CashService.php [hunk 4]
```diff
@@ -296,6 +296,27 @@ public function unlockWallet(string $walletId)
...
     }
 
+    /**
+     * @param string $walletId
+     * @param array $data
+     * @return mixed
+     */
+    public function addWalletLimit(string $walletId, array $data)
+    {
+	    return static::handleResponse($this->sendRequest()->post(static::getUrl("{$walletId}/add-limit"), [
```
Comment (dammy001) [inline src/Services/CashService.php:306]: Formatting

File: src/Http/Controllers/CPCashController.php [hunk 5]
```diff
@@ -164,6 +165,25 @@ public function unlockWallet($walletId): JsonResponse
...
+        return $this->successResponse(CPCash::addWalletLimit($walletId, $request->validated()), trans('cpcash::wallet.limit-added'));
+    }
+
+    /**
+     * @param int|string $walletId
+     * @return JsonResponse
+     */
+    public function removeWalletLimit($walletId): JsonResponse
+    {
+		return $this->successResponse(CPCash::removeWalletLimit($walletId), trans('cpcash::wallet.limit-removed'));
```
Comment (dammy001) [inline src/Http/Controllers/CPCashController.php:185]: Formatting.

**Review by dammy001 — CHANGES_REQUESTED** [review]

File: src/Http/Controllers/CPCashController.php (hunk 3 above)
Comment (Geoslim) [reply src/Http/Controllers/CPCashController.php:174]: Followed what I saw on the package.... The routes and methods and even the request classes of other ones exists in the package

Comment (dammy001) [reply src/Http/Controllers/CPCashController.php:174]: I understand. The methods in this controller are just for users only. Non-related shouldn't be added here.

Comment (Geoslim) [reply src/Http/Controllers/CPCashController.php:174]: lockWallet & unlockWallet methods were added and they are 'non-related'

Comment (dammy001) [reply src/Http/Controllers/CPCashController.php:174]: Will remove them.

**Review by dammy001 — APPROVED** [review]

**Pull request merged**